processor/src/host/advice/mod.rs

```diff
@@ -719,9 +717,8 @@ pub trait AdviceProvider: Sized {
     /// After the operation, both the original trees and the new tree remains in the advice
     /// provider (i.e., the input trees are not removed).
     ///
-    /// # Errors
-    /// Returns an error if a Merkle tree for either of the specified roots cannot be found in this
-    /// advice provider.
+    /// It is not checked whether a Merkle tree for either of the specified roots can be found in
+    /// this advice provider.
     fn merge_roots(&mut self, lhs: Word, rhs: Word) -> Result<Word, ExecutionError>;
```

Comment:
Working on this issue I noticed that _for now_ we are using this function (and `mtree_merge` instruction) only in the MMR collection. In that case may be it makes sense to rename internal functions to `merge_nodes` to avoid inconsistency in merging roots not being roots?

Reply:
I was thinking about it as well - but note sure if `merge_nodes` is better. Technically, what we are doing here is inserting a new tree into the Merkle store - so, `insert_tree` would be more appropriate, but that can also be confusing.

So, I'd probably keep it as `merge_roots`.